### DISCUSSION TITLE
How bad is it if you use a custom server?

### DISCUSSION THREAD

**Discussion by nimabrownlee** [vercel/next.js#13603] [Help]
Since unfortunately there is not a good way to setup i18n I am basically forced to use a custom server. But how bad is it to do that?

> Before deciding to use a custom server please keep in mind that it should only be used when the integrated router of Next.js can't meet your app requirements. A custom server will remove important performance optimizations, like serverless functions and Automatic Static Optimization.

The docs don't make it that comforting. Is there ways to mitigate the damage?

**Comment by lfades** [comment]
We don't recommend custom servers to keep you inside the pit of success, where everything is either static or a serverless function, you can read more about the benefits in the [JAMStack website](https://jamstack.org/).

i18n is one of our priorities for the future, it's currently not easy to implement, but we'll be working on a scalable and native implementation of it into the Next.js framework.

**Comment by leerob** [comment]
Check out the new [i18n RFC](https://github.com/vercel/next.js/discussions/17078) and give us your feedback! 🚀
